lib/firebase.dart

```diff
@@ -72,3 +74,32 @@ void _fbAuthChanged(
 
   loginCallback();
 }
+
+// Read data
+Future<model.Config> fetchConfig() async {
+  logger.debug('Fetching config from firebase ..');
+  var chartsConfigRef = firebase.firestore().doc(fb_constants.metadataPath);
+  var configSnap = await chartsConfigRef.get();
+  var configData = configSnap.data();
+
+  return model.Config.fromData(configData);
+}
+
+Future<Map<String, Map<String, dynamic>>> fetchInteractions(String path) async {
+  if (path == null || path == '') {
+    throw Error();
+  }
+
+  logger.debug('Fetching interactions from firebase ..');
+  var _interactionsRef = firebase.firestore().collection(path);
+
+  var interactionsSnap = await _interactionsRef.get();
```

Comment:
nit: similar to the other comment above, prefer `interactionsSnap -> interactionsSnapshot`

Reply:
Fixed